tests/src/data/use-cases/db-change-password.spec.ts

```diff
@@ -0,0 +1,127 @@
+import { type ChangePassword } from '@/domain/use-cases'
+import { type HashGenerator, type LoadGuardianByIdRepository, type UpdateGuardianPasswordRepository } from '@/data/protocols'
+import { DbChangePassword } from '@/data/use-cases'
+import { NotFoundError } from '@/application/errors'
+import {
+  makeFakeHashGenerator,
+  makeFakeLoadGuardianByIdRepository,
+  makeFakeUpdateGuardianPasswordRepository
+} from '@/tests/utils'
+
+interface SutTypes {
+  sut: DbChangePassword
+  hashGeneratorStub: HashGenerator
+  loadGuardianByIdRepositoryStub: LoadGuardianByIdRepository
+  updateGuardianPasswordRepositoryStub: UpdateGuardianPasswordRepository
+}
+
+const makeSut = (): SutTypes => {
+  const hashGeneratorStub = makeFakeHashGenerator()
+  const loadGuardianByIdRepositoryStub = makeFakeLoadGuardianByIdRepository()
+  const updateGuardianPasswordRepositoryStub = makeFakeUpdateGuardianPasswordRepository()
+
+  const dependencies: ChangePassword.Dependencies = {
+    hashGenerator: hashGeneratorStub,
+    loadGuardianByIdRepository: loadGuardianByIdRepositoryStub,
+    updateGuardianPasswordRepository: updateGuardianPasswordRepositoryStub
+  }
+  const sut = new DbChangePassword(dependencies)
+  return {
+    sut,
+    hashGeneratorStub,
+    loadGuardianByIdRepositoryStub,
+    updateGuardianPasswordRepositoryStub
+  }
+}
+
+describe('DbChangePassword use case', () => {
+  describe('tests HashGenerator', () => {
+    it('Should call hash generator with correct password', async () => {
+      const { sut, hashGeneratorStub } = makeSut()
+      const encryptSpy = jest.spyOn(hashGeneratorStub, 'encrypt')
+
+      const userData = { id: 'any_id', password: 'valid_password' }
```

Comment:
If you want follow GWT logic to space tests, this line must be in G part. I will try to explain my logic.

Test structure is divided in three groups. Given When Then.

Given: Setup of test, all necessary data, mock, spy, must to be defined here. 
When: Execution of sut, here you will mount, execute an recieve the returno of your feature.
Then: This is expected part of your test.